src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static Object[] ARRAY;
+    private int capacity = 10;
+    private int size = 0;
+
+    public ArrayList() {
+        ARRAY = new Object[capacity];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            ARRAY = grow();
+        }
+        ARRAY[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            ARRAY = grow();
+        }
+        System.arraycopy(ARRAY, index, ARRAY, index + 1, size - index);
+        ARRAY[index] = value;
+        size++;
+    }
 
+    private Object[] grow() {
+        capacity = capacity + capacity / 2;
+        Object[] newArray = new Object[capacity];
+        System.arraycopy(ARRAY, 0, newArray, 0, size);
+        return newArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (!list.isEmpty()) {
+            add(list.remove(0));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return (T) ARRAY[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index > size - 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        ARRAY[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removed = get(index);
```

Comment:
you have access to array, let's just check index before

Reply:
?

Reply:
?

Reply:
```suggestion
        T removed = array[index];
```
please either answer your opinion on a comment or fix it if you are totally agree. But don't ignore them